docs-ui/stories/assets/icons/details.tsx

```diff
@@ -0,0 +1,123 @@
+import {useCallback, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+import Code from 'docs-ui/components/code';
+
+import BooleanField from 'sentry/components/forms/booleanField';
+import SelectField from 'sentry/components/forms/selectField';
+import space from 'sentry/styles/space';
+
+import {iconProps} from './data';
+import IconSample from './sample';
+import {ExtendedIconData} from './searchPanel';
+
+type Props = {
+  icon: ExtendedIconData;
+};
+
+const Details = ({icon}: Props) => {
+  // Editable icon props
+  const [size, setSize] = useState(iconProps.size.default);
+  const [direction, setDirection] = useState(
+    icon.defaultProps?.direction ?? iconProps.direction.default
+  );
+  const [type, setType] = useState(icon.defaultProps?.type ?? iconProps.type.default);
+  const [isCircled, setIsCircled] = useState(icon.defaultProps?.isCircled ?? false);
+  const [isSolid, setIsSolid] = useState(icon.defaultProps?.isSolid ?? false);
+
+  /**
+   * Generate and update code sample based on prop states
+   */
+  const getCodeSample = useCallback(() => {
+    return `<Icon${icon.name} color="gray500" size="${size}"${
+      isCircled ? ' isCircled' : ' '
+    }${isSolid ? ' isSolid' : ' '} />`;
+  }, [icon.name, size, isCircled, isSolid]);
+
+  const [codeSample, setCodeSample] = useState(getCodeSample());
+
+  useEffect(() => void setCodeSample(getCodeSample()), [getCodeSample]);
```

Comment:
we can probably condense all this into a memoized variable, like:
```jsx
const codeSample = useMemo(() => …, [icon.name, size, isCircled, isSolid])
```

Reply:
oh yea, we probably can

Reply:
I'll deal with it later tho lol